sql/src/main/java/org/apache/druid/sql/calcite/aggregation/builtin/SingleValueSqlAggregator.java

```diff
@@ -0,0 +1,81 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+
+package org.apache.druid.sql.calcite.aggregation.builtin;
+
+import org.apache.calcite.rel.core.AggregateCall;
+import org.apache.calcite.sql.SqlAggFunction;
+import org.apache.calcite.sql.fun.SqlStdOperatorTable;
+import org.apache.druid.math.expr.ExprMacroTable;
+import org.apache.druid.query.aggregation.AggregatorFactory;
+import org.apache.druid.query.aggregation.SingleValueAggregatorFactory;
+import org.apache.druid.segment.column.ColumnType;
+import org.apache.druid.sql.calcite.aggregation.Aggregation;
+import org.apache.druid.sql.calcite.planner.Calcites;
+
+import javax.annotation.Nullable;
+
+public class SingleValueSqlAggregator extends SimpleSqlAggregator
+{
+
+  @Override
+  public SqlAggFunction calciteFunction()
+  {
+    return SqlStdOperatorTable.SINGLE_VALUE;
+  }
+
+  @Override
+  @Nullable
+  Aggregation getAggregation(
+      final String name,
+      final AggregateCall aggregateCall,
+      final ExprMacroTable macroTable,
+      final String fieldName
+  )
+  {
+    final ColumnType valueType = Calcites.getColumnTypeForRelDataType(aggregateCall.getType());
+    if (valueType == null) {
+      return null;
+    }
+    return Aggregation.create(createSingleValueAggregatorFactory(
+        valueType,
+        name,
+        fieldName
+    ));
+  }
+
+  static AggregatorFactory createSingleValueAggregatorFactory(
+      final ColumnType aggregationType,
+      final String name,
+      final String fieldName
+  )
+  {
+    switch (aggregationType.getType()) {
+      case LONG:
+      case FLOAT:
+      case DOUBLE:
+      case STRING:
+        return new SingleValueAggregatorFactory(name, fieldName, aggregationType);
+      default:
+        // This error refers to the Druid type. But, we're in SQL validation.
+        // It should refer to the SQL type.
+        throw SimpleSqlAggregator.badTypeException(fieldName, "SINGLE_VALUE", aggregationType);
+    }
```

Comment:
i think i already asked this, but is this needed or can it just always make a `SingleValueAggregatorFactory` with any `aggregationType`?